### PR TITLE
Fresh(#311)

### PR DESCRIPTION
**Pull request by bestgopher**
#311, copied from [jshttp/fresh](https://github.com/jshttp/fresh/blob/10e0471669dbbfbfd8de65bc6efac2ddd0bfa057/index.js#L33)

### CODE REVIEW

**Comment by Fenny** [comment]
Awesome work @bestgopher, thanks for helping us out!

But since it's a new feature, it is required to be added to the `ctx_test.go` for testing.
Do you have time to add a test case?

File: utils.go [hunk 1]
```diff
@@ -156,6 +156,34 @@ func (c *testConn) SetDeadline(t time.Time) error      { return nil }
 func (c *testConn) SetReadDeadline(t time.Time) error  { return nil }
 func (c *testConn) SetWriteDeadline(t time.Time) error { return nil }
 
+// Copied from:
```
Comment (jozsefsallai) [inline utils.go:159]: "copied from" sounds kinda bad. I'd replace it with something like "based on" or "inspired by" or maybe even "adapted from"

Comment (bestgopher) [reply utils.go:159]: I am sorry that I am not good at English, so there may be some words that are not appropriate.

And these codes are not tested.

Comment (Fenny) [reply utils.go:159]: Could you pplease change it to:
`Adapted from https://github.com/jshttp/fresh`

**Pull request merged**